xarray/backends/api.py

```diff
@@ -318,6 +318,33 @@ def maybe_decode_store(store, lock=False):
     return maybe_decode_store(store)
 
 
+def open_rasterio(filename, add_latlon=False):
+    """Open a file with RasterIO (experimental).
+
+    This should work with any file that rasterio can open (typically: 
+    geoTIFF). The x and y coordinates are generated automatically out of the 
+    file's geoinformation.
+
+    Parameters
+    ----------
+    filename : str
+        path to the file to open
+    add_latlon : bool, optional
+        if the file ships with valid geoinformation, longitudes and latitudes
+        can be computed and added to the dataset as non-dimension coordinates
+
+    Returns
+    -------
+    dataset : Dataset
+        The newly created dataset.
+    """
+    ds = conventions.decode_cf(backends.RasterioDataStore(filename))
+    if add_latlon:
+        from ..core.utils import add_latlon_coords_from_crs
+        ds = add_latlon_coords_from_crs(ds)
```

Comment:
Should we detect an empty CRS here? Or are we happy with the current behavior (shown below)

```
In [49]: ds = xr.open_rasterio('ssp1_g.tiff', add_latlon=True)
---------------------------------------------------------------------------
CRSError                                  Traceback (most recent call last)
<ipython-input-49-6f687cb4abd7> in <module>()
----> 1 ds = xr.open_rasterio('ssp1_g.tiff', add_latlon=True)

/home/gidden/.local/lib/python2.7/site-packages/xarray-0.9.5_34_g48c7268-py2.7.egg/xarray/backends/api.pyc in open_rasterio(filename, add_latlon)
    342     if add_latlon:
    343         from ..core.utils import add_latlon_coords_from_crs
--> 344         ds = add_latlon_coords_from_crs(ds)
    345     return ds
    346 

/home/gidden/.local/lib/python2.7/site-packages/xarray-0.9.5_34_g48c7268-py2.7.egg/xarray/core/utils.pyc in add_latlon_coords_from_crs(ds, crs)
    520     # Rasterio works with 1D arrays
    521     xc, yc = transform(crs, {'init': 'EPSG:4326'},
--> 522                        x.flatten(), y.flatten())
    523     xc = np.asarray(xc).reshape((ny, nx))
    524     yc = np.asarray(yc).reshape((ny, nx))

/home/gidden/.local/lib/python2.7/site-packages/rasterio/warp.pyc in transform(src_crs, dst_crs, xs, ys, zs)
     50     coordinate reference system.
     51     """
---> 52     return _transform(src_crs, dst_crs, xs, ys, zs)
     53 
     54 

rasterio/_base.pyx in rasterio._base._transform (rasterio/_base.c:19011)()

CRSError: No translation for an empty SRS to PROJ.4 format is known.
```